os/binfmt/libelf/libelf_addrenv.c

```diff
@@ -229,7 +240,11 @@ int elf_addrenv_alloc(FAR struct elf_loadinfo_s *loadinfo)
 #elif defined(CONFIG_ARMV8M_MPU)
 	loadinfo->binp->ramsize = MPU_ALIGN_UP(loadinfo->binp->ramsize);
 	loadinfo->binp->ramstart = (uint32_t)kmm_memalign_at(CONFIG_HEAP_INDEX_LOADED_APP, MPU_ALIGNMENT_BYTES, loadinfo->binp->ramsize);
+#elif defined(CONFIG_ARCH_USE_MMU)
+	loadinfo->binp->ramsize = MMU_ALIGN_UP(loadinfo->binp->ramsize);
+	loadinfo->binp->ramstart = (uint32_t)kmm_memalign_at(CONFIG_HEAP_INDEX_LOADED_APP, MMU_ALIGNMENT_BYTES, loadinfo->binp->ramsize);
```

Comment:
Is it possible to move architecture-specific codes in a new file?
I think we should keep common files and arch-specific files separately.
Mixed code is much difficult to maintain.

Reply:
In this case, it will be difficult to separate arch code. The memory for apps needs to be allocated based on the version of mmu or mpu used. But also, the allocated memory needs to be stored in the various data structures related to binary loader.

Reply:
In common file
```
int abc(void)
{
    ....
    loadinfo->binp->ramstart = boardspecific_memory(loadinfo);
    ...
}
```
In board-specific file
```
void *boardspecific_memory(.....)
{
#ifdef ARMV7_MPU
     .....
#elif ARMV8M_MPU
    ....
#elif ARMV7A_MMU
    ...
#else
    ...
#endif
}
```

Impossible this kind of separation?

Reply:
Both can be in binfmt, but I want to separate a file for common and board-specific code like ARMv7m, v8m and v7a or mmu and mpu.

Reply:
@kishore-sn Could you let me know your opinion for this?

Reply:
Done. Pls check.